tests/core/test_log_path_timetag.py

```
@@ -0,0 +1,224 @@
+        with Pds3File._pinned_log_timetag():
+            inside = pdsf.log_path_for_bundle('_md5', task='validate', dir='tool')
+
+        assert Pds3File._LOG_TIMETAG is None
+        after = [pdsf.log_path_for_bundle('_md5', task='validate', dir='tool'),
+                 pdsf.log_path_for_bundle('_md5', task='validate', dir='tool')]
+        assert len(timetags_of(after) | timetags_of([inside])) == 3
+
+    def test_the_pin_is_released_when_the_block_raises(self, ticking_clock):
+        """A task that raises inside the block must not leave the tag pinned."""
+
+        with pytest.raises(ValueError), Pds3File._pinned_log_timetag():
```

Ruff (0.16.1)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)